### ISSUE TITLE
msmbuilder/example_datasets/_muller.c

### ISSUE THREAD

**Issue by akiikius** [msmbuilder/msmbuilder#475]
Hi,

Am trying to install msmbuilder but i keep getting this error:

clang: error: no such file or directory: 'msmbuilder/example_datasets/_muller.c'
clang: error: no input files
error: command 'cc' failed with exit status 1

What could be the problem? I have installed all the dependencies.

Sylvester

**Comment by mpharrigan** [comment]
How are you installing? `pip`? `python setup.py install`?

**Comment by akiikius** [comment]
This is how:

python setup.py install

**Comment by rmcgibbo** [comment]
So the file `msmbuilder/example_datasets/_muller.c` should be generate during the installation by a tool called cython, but evidently this is not happening

Can you run the following command on the command line, and paste the result? This will help us track down which step is going wrong.

1. `python --version`
2. `which python`
3. `cython --version`
4. `which cython`
5. `uname -a`

**Comment by akiikius** [comment]
Sylvesters-Mac-mini:msmbuilder stumusii$ python --version
Python 2.7.5
Sylvesters-Mac-mini:msmbuilder stumusii$ which python
/usr/bin/python
Sylvesters-Mac-mini:msmbuilder stumusii$ cython --version
Cython version 0.22
Sylvesters-Mac-mini:msmbuilder stumusii$ which cython
/usr/local/bin/cython
Sylvesters-Mac-mini:msmbuilder stumusii$ uname -a
Darwin Sylvesters-Mac-mini.local 13.3.0 Darwin Kernel Version 13.3.0: Tue Jun  3 21:27:35 PDT 2014; root:xnu-2422.110.17~1/RELEASE_X86_64 x86_64

**Comment by rmcgibbo** [comment]
Can you also paste the entire output of python setup.py install?

Off the top of my head I'm not totally sure what the problem is. Likely
cython is not installed correctly.

I would recommend following our instructions on mdtraj.org and using
anaconda python, instead of the one that came preinstalled with the OS, if
that's an option for you.

**Comment by akiikius** [comment]
Sylvesters-Mac-mini:msmbuilder stumusii$ sudo python setup.py install
Password:
C compiler:
Apple LLVM version 6.0 (clang-600.0.56) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.3.0
Thread model: posix
Attempting to autodetect OpenMP support... Did not detect OpenMP support
Attempting to autodetect SSE3   support... Compiler supports SSE3
Attempting to autodetect SSE4.1 support... Compiler supports SSE4.1

running install
running bdist_egg
running egg_info
writing msmbuilder.egg-info/PKG-INFO
writing top-level names to msmbuilder.egg-info/top_level.txt
writing dependency_links to msmbuilder.egg-info/dependency_links.txt
writing entry points to msmbuilder.egg-info/entry_points.txt
reading manifest file 'msmbuilder.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'msmbuilder.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.9-intel/egg
running install_lib
running build_py
copying msmbuilder/version.py -> build/lib.macosx-10.9-intel-2.7/msmbuilder
running build_ext
building 'msmbuilder.example_datasets._muller' extension
cc -fno-strict-aliasing -fno-common -dynamic -arch x86_64 -arch i386 -g -Os -pipe -fno-common -fno-strict-aliasing -fwrapv -DENABLE_DTRACE -DMACOSX -DNDEBUG -Wall -Wstrict-prototypes -Wshorten-64-to-32 -DNDEBUG -g -fwrapv -Os -Wall -Wstrict-prototypes -DENABLE_DTRACE -arch x86_64 -arch i386 -pipe -I/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/core/include -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c msmbuilder/example_datasets/_muller.c -o build/temp.macosx-10.9-intel-2.7/msmbuilder/example_datasets/_muller.o
clang: error: no such file or directory: 'msmbuilder/example_datasets/_muller.c'
clang: error: no input files
error: command 'cc' failed with exit status 1

**Comment by kyleabeauchamp** [comment]
I strongly recommend using Anaconda python, rather than system python and `sudo`

**Comment by akiikius** [comment]
Okay, will do, Thanks for your help.

**Comment by akiikius** [comment]
One final question, will Anaconda and msmbuilder have to be installed in the same directory?

**Comment by kyleabeauchamp** [comment]
If you call `python setup.py install` using the correct `python` executable from Anaconda, you should not have to worry about which directory--anaconda python knows to install msmbuilder into the right place.

**Comment by akiikius** [comment]
Hi,

I installed anaconda and re-attempted to install msmbuilder as per the instructions on the msmbuilder website. I get this error:

building 'msmbuilder.example_datasets._muller' extension
creating build/temp.macosx-10.9-intel-2.7
creating build/temp.macosx-10.9-intel-2.7/msmbuilder
creating build/temp.macosx-10.9-intel-2.7/msmbuilder/example_datasets
cc -fno-strict-aliasing -fno-common -dynamic -arch x86_64 -arch i386 -g -Os -pipe -fno-common -fno-strict-aliasing -fwrapv -DENABLE_DTRACE -DMACOSX -DNDEBUG -Wall -Wstrict-prototypes -Wshorten-64-to-32 -DNDEBUG -g -fwrapv -Os -Wall -Wstrict-prototypes -DENABLE_DTRACE -arch x86_64 -arch i386 -pipe -I/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/core/include -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c msmbuilder/example_datasets/_muller.c -o build/temp.macosx-10.9-intel-2.7/msmbuilder/example_datasets/_muller.o
clang: error: no such file or directory: 'msmbuilder/example_datasets/_muller.c'
clang: error: no input files
error: command 'cc' failed with exit status 1


I installed as follows:

Sylvesters-Mac-mini:Downloads stumusii$ cd msmbuilder/
Sylvesters-Mac-mini:msmbuilder stumusii$ sudo python setup.py install

**Comment by kyleabeauchamp** [comment]
Try the following?

```
conda config --add channels http://conda.binstar.org/omnia/
conda install msmbuilder
```

**Comment by akiikius** [comment]
It seems to have worked with the instructions you sent (see below). Am going to try and do an analysis to test.
Thanks!


$ conda config --add channels http://conda.binstar.org/omnia/
$ conda install msmbuilder
Fetching package metadata: ...
Solving package specifications: .
Package plan for installation in environment /Users/stumusii/anaconda:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-3.9.0                |           py27_0         163 KB
    conda-env-2.1.3            |           py27_0          15 KB
    hdf5-1.8.14                |                0         1.5 MB
    mdtraj-1.2.0               |       np19py27_0         5.4 MB
    msmbuilder-3.0.0           |       np19py27_0         814 KB
    numpy-1.9.1                |           py27_0         2.9 MB
    numpydoc-0.5               |           py27_0         1.0 MB
    openssl-1.0.1k             |                0         2.5 MB
    pandas-0.15.2              |       np19py27_0         4.7 MB
    pytables-3.1.1             |       np19py27_2         1.4 MB
    python-2.7.9               |                1        11.3 MB
    pytz-2014.9                |           py27_0         175 KB
    requests-2.5.3             |           py27_0         592 KB
    scipy-0.15.1               |       np19py27_0        12.1 MB
    scripttest-1.3             |           py27_0          17 KB
    setuptools-12.2            |           py27_0         437 KB
    six-1.9.0                  |           py27_0          16 KB
    zlib-1.2.8                 |                0          83 KB
    ------------------------------------------------------------
                                           Total:        45.1 MB

The following NEW packages will be INSTALLED:

    conda-env:  2.1.3-py27_0     
    mdtraj:     1.2.0-np19py27_0 
    msmbuilder: 3.0.0-np19py27_0 
    numpydoc:   0.5-py27_0       
    scripttest: 1.3-py27_0       

The following packages will be UPDATED:

    conda:      3.7.0-py27_0      --> 3.9.0-py27_0     
    hdf5:       1.8.13-0          --> 1.8.14-0         
    numpy:      1.9.0-py27_0      --> 1.9.1-py27_0     
    openssl:    1.0.1h-1          --> 1.0.1k-0         
    pandas:     0.14.1-np19py27_0 --> 0.15.2-np19py27_0
    pytables:   3.1.1-np19py27_0  --> 3.1.1-np19py27_2 
    python:     2.7.8-1           --> 2.7.9-1          
    pytz:       2014.7-py27_0     --> 2014.9-py27_0    
    requests:   2.4.1-py27_0      --> 2.5.3-py27_0     
    scipy:      0.14.0-np19py27_0 --> 0.15.1-np19py27_0
    setuptools: 5.8-py27_0        --> 12.2-py27_0      
    six:        1.8.0-py27_0      --> 1.9.0-py27_0     
    zlib:       1.2.7-1           --> 1.2.8-0          

Proceed ([y]/n)? y

Fetching packages ...
conda-3.9.0-py 100% |##############################################################################################################################################################################| Time: 0:00:00 654.64 kB/s
conda-env-2.1. 100% |##############################################################################################################################################################################| Time: 0:00:00  88.56 kB/s
hdf5-1.8.14-0. 100% |##############################################################################################################################################################################| Time: 0:00:01   1.37 MB/s
mdtraj-1.2.0-n 100% |##############################################################################################################################################################################| Time: 0:00:05   1.01 MB/s
msmbuilder-3.0 100% |##############################################################################################################################################################################| Time: 0:00:00   1.47 MB/s
numpy-1.9.1-py 100% |##############################################################################################################################################################################| Time: 0:00:01   2.94 MB/s
numpydoc-0.5-p 100% |##############################################################################################################################################################################| Time: 0:00:00   1.70 MB/s
openssl-1.0.1k 100% |##############################################################################################################################################################################| Time: 0:00:01   1.38 MB/s
pandas-0.15.2- 100% |##############################################################################################################################################################################| Time: 0:00:01   3.85 MB/s
pytables-3.1.1 100% |##############################################################################################################################################################################| Time: 0:00:00   1.91 MB/s
python-2.7.9-1 100% |##############################################################################################################################################################################| Time: 0:00:12 980.36 kB/s
pytz-2014.9-py 100% |##############################################################################################################################################################################| Time: 0:00:00 620.93 kB/s
requests-2.5.3 100% |##############################################################################################################################################################################| Time: 0:00:03 165.11 kB/s
scipy-0.15.1-n 100% |##############################################################################################################################################################################| Time: 0:00:04   2.72 MB/s
scripttest-1.3 100% |##############################################################################################################################################################################| Time: 0:00:00 305.00 kB/s
setuptools-12. 100% |##############################################################################################################################################################################| Time: 0:00:01 412.10 kB/s
six-1.9.0-py27 100% |##############################################################################################################################################################################| Time: 0:00:00 322.06 kB/s
zlib-1.2.8-0.t 100% |##############################################################################################################################################################################| Time: 0:00:00 252.22 kB/s
Extracting packages ...
[      COMPLETE      ] |################################################################################################################################################################################################| 100%
Unlinking packages ...
[      COMPLETE      ] |################################################################################################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ] |################################################################################################################################################################################################| 100%

**Comment by akiikius** [comment]
Tried to test it: $ msmb -h

Gives me this error:

-bash: msmb: command not found

**Comment by kyleabeauchamp** [comment]
Can you do the following two tests:

 `which python`

```
ipython
import msmbuilder as msmb
```

**Comment by akiikius** [comment]
stumusii$ which python
/usr/bin/python
Sylvesters-Mac-mini:DIVALENTS stumusii$ ipython
Python 2.7.8 (default, Jul 13 2014, 17:11:32) 
Type "copyright", "credits" or "license" for more information.

IPython 2.2.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: 

In [1]: import msmbuilder as msmb
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-433fde979c33> in <module>()
----> 1 import msmbuilder as msmb

ImportError: No module named msmbuilder

**Comment by kyleabeauchamp** [comment]
You need to be using the anaconda python executable, not the system one.  You installed msmbuilder to your anaconda python.

**Comment by kyleabeauchamp** [comment]
Your PATH environment variable needs to be fixed.

**Comment by akiikius** [comment]
Is ConvertDataToHDF not a part of msmbuilder anymore?

when i type: msmb -h, i get a list that does not include ConvertDataToHDF. 

thanks!

See below:

$ msmb -h

usage: msmb [-h]  ...

Statistical models for biomolecular dynamics

optional arguments:
 -h, --help             show this help message and exit

commands:
 
  AtomIndices           Create index file for atoms or distance pairs.
  ConvertChunkedProject
                        Convert an MD dataset with chunked trajectories into a standard format.
  DihedralFeaturizer    Featurizer based on dihedral angles.
  AtomPairsFeaturizer   Featurizer based on distances between specified pairs of atoms.
  SuperposeFeaturizer   Featurizer based on euclidian atom distances to reference structure.
  DRIDFeaturizer        Featurizer based on distribution of reciprocal interatomic distances (DRID).
  ContactFeaturizer     Featurizer based on residue-residue distances.
  KMeans                K-Means clustering.
  KCenters              K-Centers clustering.
  KMedoids              K-Medoids clustering.
  MiniBatchKMedoids     Mini-Batch K-Medoids clustering.
  RegularSpatial        Regular spatial clustering.
  MiniBatchKMeans       Mini-Batch K-Means clustering.
  tICA                  Time-structure Independent Component Analysis (tICA).
  SparseTICA            Sparse time-structure Independent Component Analysis (tICA).
  PCA                   Principal component analysis (PCA).
  AlanineDipeptide      Download example alanine dipeptide dataset.
  FsPeptide             Download example Fs-peptide dataset.
  MetEnkephalin         Download example Met-Enkephalin dataset.
  DoubleWell            Generate example double well potential dataset.
  QuadWell              Generate example quad-well potential dataset.
  ImpliedTimescales     Scan the implied timescales of MarkovStateModels with respect to lag time.
  GaussianFusionHMM     Reversible Gaussian Hidden Markov Model L1-Fusion Regularization.
  MarkovStateModel      Reversible Markov State Model.
  BayesianMarkovStateModel
                        Bayesian Markov state model.
  ContinuousTimeMSM     Reversible first order master equation model.
  TransformDataset      Use a pre-fit model to transform a dataset.

**Comment by kyleabeauchamp** [comment]
Not a part of MSMB3

**Comment by kyleabeauchamp** [comment]
Look into ConvertChunkedProject

**Comment by akiikius** [comment]
Tried a run of Kmedoids:
stumusii$ msmb KMedoids  -i cab-dry.xtc -o /Users/stumusii/Desktop/MARKOV-MODELLING/xtc/OUTPUT -t /Users/stumusii/Desktop/MARKOV-MODELLING/xtc/TRANSFORMED --stride 1 --top /Users/stumusii/Desktop/MARKOV-MODELLING/xtc/CaB-run/cab-dry.prmtop --atom_indices Atom.txt --n_clusters 20 --n_passes 1 --metric rmsd --random_state 1
KMedoids(metric='rmsd', n_clusters=20, n_passes=1, random_state=1)
Fitting model...

And got following output:

An unexpected error has occurred with MSMBuilder (version 3.0.0), please
consider sending the following traceback to MSMBuilder GitHub issue tracker at:
            https://github.com/msmbuilder/msmbuilder/issues

Traceback (most recent call last):
  File "/Users/stumusii/anaconda/envs/py27/bin/msmb", line 6, in <module>
    sys.exit(main())
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/scripts/msmb.py", line 28, in main
    app.start()
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/cmdline.py", line 415, in start
    instance.start()
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/commands/fit_transform.py", line 58, in start
    self.instance.fit(inp_ds)
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/cluster/kmedoids.py", line 155, in fit
    MultiSequenceClusterMixin.fit(self, sequences)
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/cluster/base.py", line 47, in fit
    check_iter_of_sequences(sequences, allow_trajectory=self._allow_trajectory)
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/utils/validation.py", line 43, in check_iter_of_sequences
    for i, X in enumerate(sequences):
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/dataset.py", line 177, in __iter__
    yield self.get(key)
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/dataset.py", line 383, in get
    atom_indices=self.atom_indices, top=self._topology)
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/mdtraj/core/trajectory.py", line 362, in load
    value = loader(filename, **kwargs)
  File "xtc.pyx", line 141, in xtc.load_xtc (MDTraj/formats/xtc/xtc.c:2370)
  File "xtc.pyx", line 146, in xtc.load_xtc (MDTraj/formats/xtc/xtc.c:2182)
  File "xtc.pyx", line 349, in xtc.XTCTrajectoryFile.read (MDTraj/formats/xtc/xtc.c:4617)
IndexError: invalid slice

**Comment by rmcgibbo** [comment]
I can reproduce this. Fix coming.

**Comment by rmcgibbo** [comment]
Thanks for reporting that bug. It's been fixed in the development branch. In the mean time, you can work around it by simply not supplying the `--stride` option. 1 is the default anyways.

**Comment by akiikius** [comment]
1. I Saw recommendation to remove stride,

2. Removed stride and run as below:

$ msmb KMedoids  -i /Users/stumusii/Desktop/xtc/CaB-run/cab-dry.xtc -o /Users/xtc/OUTPUT/ -t /Desktop/xtc/TRANSFORMED/  --top /Users/stumusii/Desktop/xtc/CaB-run/dry.prmtop --atom_indices Atom.dat --n_clusters 20 --n_passes 1 --metric rmsd --random_state 1
KMedoids(metric='rmsd', n_clusters=20, n_passes=1, random_state=1)
Fitting model...

3. Got error:

An unexpected error has occurred with MSMBuilder (version 3.0.0), please
consider sending the following traceback to MSMBuilder GitHub issue tracker at:
            https://github.com/msmbuilder/msmbuilder/issues

Traceback (most recent call last):
  File "/Users/stumusii/anaconda/envs/py27/bin/msmb", line 6, in <module>
    sys.exit(main())
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/scripts/msmb.py", line 28, in main
    app.start()
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/cmdline.py", line 415, in start
    instance.start()
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/commands/fit_transform.py", line 58, in start
    self.instance.fit(inp_ds)
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/cluster/kmedoids.py", line 155, in fit
    MultiSequenceClusterMixin.fit(self, sequences)
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/cluster/base.py", line 48, in fit
    super(MultiSequenceClusterMixin, self).fit(self._concat(sequences))
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/cluster/base.py", line 65, in _concat
    concat = sequences[0].join(sequences[1:])
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/dataset.py", line 170, in __getitem__
    return self.get(i)
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/msmbuilder-3.0.0-py2.7-macosx-10.5-x86_64.egg/msmbuilder/dataset.py", line 383, in get
    atom_indices=self.atom_indices, top=self._topology)
  File "/Users/stumusii/anaconda/envs/py27/lib/python2.7/site-packages/mdtraj/core/trajectory.py", line 325, in load
    raise ValueError('No trajectories specified. '
ValueError: No trajectories specified. filename_or_filenames was an empty list

4.  My trajectory file looks fine.

**Comment by rmcgibbo** [comment]
Sorry about that. This was fixed in master with #479.

**Comment by rmcgibbo** [comment]
@kyleabeauchamp, @mpharrigan: we should discuss the release timeline.
